src/pages/AdminPages/Transactions/PendingTransactions.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import BaseModal from '../../../components/Modal/BaseModal'
+import { Table } from '../../../components/Table'
+import { setSelectedAlterContent, sendTransactionResponse } from '../../../store/reducers/adminQuery'
+import { Form, SubmitButton, TextareaField } from '../../../components/Form'
+import { disableButton } from '../../../store/reducers/buttonState'
+import * as Yup from 'yup'
+import { thousands_separators, timeAgo, TIME_ZONE } from '../../../utils/helperFunctions'
+
+
+const validationSchema = Yup.object().shape({
+    reason: Yup.string().required().min(3).label("Declination Reason")
+})
+
+
+const PendingTransactions = ({ 
+    pendingTransactions,
+    processedTransactions,
+    selectedAlterContent,
+    buttonDisable,
+    header,
+    setContent
+}) => {
+
+
+    const [showResponse, setShowResponse] = useState(false)
+
+    const dispatch = useDispatch()
+
+    const closeShowResponse = () => {
+        dispatch(setSelectedAlterContent({}))
+        setShowResponse(false)
+    }
+
+
+    const processTransaction = ({ reason, status }) => {
+
+        const transactionResponse = {
+            processedTransactions,
+            pendingTransactions,
+            data : {
+                ...selectedAlterContent,
+                reason,
+                status
+            }
+        }
+
+        dispatch(sendTransactionResponse(transactionResponse))
+        dispatch(disableButton())
```

Comment:
The order of this should be reversed. The button should be disabled immediately the button is clicked and calling ```dispatch(sendTransactionResponse(transactionResponse))``` should enable the button back.

Reply:
Oh... okay... would change that